### DISCUSSION TITLE
a few private questions about setting up a json:scada for production

### DISCUSSION THREAD

**Discussion by joyshmitz** [riclolsen/json-scada#15] [Q&A] [answered]
1. Ability to switch layers in the display view
2. in the table of signals of points that the parameter of a normal mode of each point is displayed. (preset for comparison)
3. How to configure the transition from alarm (or tooltips) events to the desired SVG file
4. whether there is a generator of reports / printed forms, or which one can be applied in your opinion

**Comment by riclolsen** [comment, marked as answer]
1. You can create layers in Inkscape and toggle them visible/invisible by using Javascript (tied to a button event, for example).
2. This can be useful. The property "valueDefault" can be used for comparison, but you need to create a separate report. I may integrate this feature in the future in Tabular Viewer. A new property like "valueNormal" may be added.
3. Not possible now. This is also useful for future improvements. This may require the creation of a new table or some kind of rule to link groups of points to displays.
4. There is no integrated report generator. I usually create simple reports via PHP/HTML tables. I can suggest Metabase, Helicalinsight and Widestage that can possibly work well with Mongo and Postgresql. If you can find others, please suggest.

**Reply by joyshmitz** [reply]
next questions 
where is the log of user actions, they are recorded in the database, but this is not in the interface

**Reply by riclolsen** [reply]
There is no interface yet, just logged to userActions mongo collection.

**Reply by joyshmitz** [reply]
another q
 How to set point value manually by script from .svg file and take this value visible in event table as manual changed?
you can add this function from here (pictures example) https://drive.google.com/file/d/1fZNi6y--K5kEkBNM54zI6QCIzUGuad--/view?usp=sharing

**Reply by riclolsen** [reply]
Manual values entered now generate events for Events Viewer. Also for annotations.
Better userActions logs (only modified properties registered).
https://github.com/riclolsen/json-scada/blob/master/src/server_realtime_auth/index.js

**Reply by joyshmitz** [reply]
And how to change the point manually from it. What functions can be used and how to  access to point value.

We need also some function to block this value from changes, e.g. need to change the point value and block it from other changes untill we don't cancel this block.

How to set up the point to get ability of this command function?
https://drive.google.com/file/d/16bRqoBwjUR187qufD3D-l3zNCJqe1CwI/view?usp=sharing
Because I can't see it at my screens

**Reply by riclolsen** [reply]
At this moment only points defined as origin=manual can have value altered by the operator.  This will be changed in the future to allow changes on any point provided the user has rights. 
There is not a ready function to write values currently. It is done via a Post call, see here
https://github.com/riclolsen/json-scada/blob/92a98bcd5c41efdc1a1deda022212521e79e14b7/src/htdocs/websage.js#L1503
I will provide a call that can be used in SVG in a future version for this purpose.

To have command linked, a supervised point must have a property 'commandOfSupervised' set to the point '_id' of the command point and the command point must have a 'supervisedOfCommand' property set in the reverse direction. See demo database to see examples.

**Reply by riclolsen** [reply]
User actions were added to Admin UI.

**Reply by riclolsen** [reply]
I have tried Widestage, Jasper Reports and Knowage with mongo and have no luck with any of them. I think with Postgresql it should be all ok however. If anyone can find some useful reporting tool, please report here.
